service/get.go

```diff
@@ -0,0 +1,80 @@
+package service
+
+import (
+	"fmt"
+	"sync"
+
+	"github.com/openSUSE/helm-mirror/utils"
+	yaml "gopkg.in/yaml.v2"
+)
+
+// GetServiceInterface defines a Get service
+type GetServiceInterface interface {
+	Get() error
+}
+
+// GetService structure definition
+type GetService struct {
+	helmRepo string
+	logger   utils.Logger
+	io       utils.Io
+	web      utils.Web
+}
+
+// NewGetService return a new instace of GetService
+func NewGetService(helmRepo string, logger utils.Logger, io utils.Io, web utils.Web) GetServiceInterface {
+	return &GetService{
+		helmRepo: helmRepo,
+		logger:   logger,
+		io:       io,
+		web:      web,
+	}
+}
+
+//Get methods downloads the index file and the Helm charts to the working directory.
+func (g *GetService) Get() error {
+	b, err := g.web.DownloadFile(g.helmRepo + "/index.yaml")
+
+	if err != nil {
+		return err
+	}
+	err = g.io.WriteFile("index.yaml", b)
+	if err != nil {
+		return err
+	}
+	var charts Charts
+	err = yaml.Unmarshal(b, &charts)
+	if err != nil {
+		g.logger.Error("unmarshaling charts: %s", err)
+		return err
+	}
+
+	var wg sync.WaitGroup
+	for _, charts := range charts.Entries {
+		for _, c := range charts {
+			fileName := fmt.Sprintf("%s-%s.tgz", c.Name, c.Version)
+			for _, u := range c.Urls {
+				wg.Add(1)
+				go g.downloadFile(u, fileName, &wg)
```

Comment:
If you already have the file downloads you shouldn't need to do it a second time. What about only downloading the new or changed charts?

The digest for a chart can be used as a unique content ID to look for changes and check what you already have.